build.sbt

```diff
@@ -232,26 +240,56 @@ lazy val allAggregates = projectsWithOptionalNative ++
 
 // For CI tests, defining scripts that run JVM/JS/Native tests separately
 val testJVM = taskKey[Unit]("Test JVM projects")
-val testJS1 = taskKey[Unit]("Test JS 2.11 and 2.12 projects")
-val testJS2 = taskKey[Unit]("Test JS 2.13 and 3 projects")
+val testJS2_11 = taskKey[Unit]("Test JS 2.11 projects")
+val testJS2_12 = taskKey[Unit]("Test JS 2.12 projects")
+val testJS2_13 = taskKey[Unit]("Test JS 2.13 projects")
+val testJS3 = taskKey[Unit]("Test JS 3 projects")
 val testNative = taskKey[Unit]("Test native projects")
 
 def filterProject(p: String => Boolean) =
   ScopeFilter(inProjects(allAggregates.filter(pr => p(display(pr.project))): _*))
 
+def filterByVersionAndPlatform(scalaVersionFilter: String, platformFilter: String) = filterProject { projectName =>
+  val byPlatform =
+    if (platformFilter == "JVM") !projectName.contains("JS") && !projectName.contains("Native")
+    else projectName.contains(platformFilter)
+  val byVersion = scalaVersionFilter match {
+    case "3"    => projectName.contains("3")
+    case "2.13" => projectName.contains("2_13")
+    case "2.12" => projectName.contains("2_12")
+    case "2.11" => projectName.contains("2_11")
+  }
+
+  byPlatform && byVersion
+}
+
 lazy val rootProject = (project in file("."))
   .settings(commonSettings: _*)
   .settings(
     publish / skip := true,
     name := "sttp",
     testJVM := (Test / test).all(filterProject(p => !p.contains("JS") && !p.contains("Native"))).value,
-    testJS1 := (Test / test)
-      .all(filterProject(p => p.contains("JS") && (p.contains("2_11") || p.contains("2_12"))))
+    testJS2_11 := (Test / test)
```

Comment:
how does this differ from `testScoped`?

Reply:
JS tests are run in different way then others there is a comment in ci file, it works the same way in tapir

Reply:
Hm which comment?